crates/ruff/resources/test/fixtures/flake8_pyi/PYI016.pyi

```diff
@@ -74,3 +74,13 @@ field22: typing.Union[int, typing.Union[int, typing.Union[int, int]]]  # Error
 # Should emit in cases with newlines
 field23: set[  # foo
     int] | set[int]
+
+# Should emit twice (once for each `int` in the nested union, both of which are
+# duplicates of the outer `int`), but not three times (which would indicate that
+# we incorrectly re-checked the nested union).
+field24: typing.Union[int, typing.Union[int, int]]  # PYI016: Duplicate union member `int`
```

Comment:
Prior to https://github.com/astral-sh/ruff/pull/6345, we emitted three times here (incorrectly). See: https://github.com/astral-sh/ruff/issues/6285.